integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -396,6 +399,87 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
     });
   });
 
+  describe("Errors", () => {
+    it("raises compensating writes errors", async function () {
+      const person1Id = new BSON.ObjectId("0000002a9a7969d24bea4cf5");
+      const person2Id = new BSON.ObjectId("0000002a9a7969d24bea4cf6");
+      const dogId = new BSON.ObjectId("0000002a9a7969d24bea4cf7");
+
+      const promise = new Promise<void>((resolve, _) => {
+        (async () => {
```

Comment:
Why the inner body? This could result in an unhandled promise rejection.

```suggestion
      const promise = new Promise<void>(async (resolve, _) => {
```

Reply:
I agree on awaiting right away, I will fix it.

Regarding why the inner body, I did it because of this linting rule: https://eslint.org/docs/latest/rules/no-async-promise-executor. I was reading that having it defined as async could cause some issue, like the fact that if it throws, then the promise will never be resolved. After doing some testing I think that this probably behaves in the same way as the version that the linter complains about. 

What do you think, should I just ignore the linter or do you have any other suggestion about how to do this?